src/collect/collect.c

```diff
@@ -112,7 +116,8 @@ static int prepare(char *dir, char *filename)
                         fprintf(stderr, "Lock taken, wait for %d seconds\n", UDEV_ALARM_TIMEOUT);
                 if (errno == EAGAIN || errno == EACCES) {
                         alarm(UDEV_ALARM_TIMEOUT);
-                        lockf(fd, F_LOCK, 0);
+                        if (lockf(fd, F_LOCK, 0)) { /* TODO: implement proper error handling */
+                        }
```

Comment:
we should just print the error and fail the function here (though faling this would be quite the edge case, not even sure it's possible in practice, but regardless)

Reply:
Now I see that `prepare` tries to lock even when `open` returned a negative fd. I will add a separate commit addressing that.

Reply:
As I understand what this code does, it first tries to do a non-blocking lock, and in case it fails, it retries a blocking one and sets a timeout (by alarm). But in case the non-blocking lock fails with something different from `EAGAIN` or `EACCES` then `prepare` will return an unlocked fd. I will change it to fail in the later case too and also make the error not guarded by `if (debug)`.